post-processor.go

```diff
@@ -84,16 +87,36 @@ func (p *PostProcessor) Configure(raws ...interface{}) error {
 	}
 
 	var cred *credentials.Credentials = nil // nil credentials use the default aws sdk credential chain
-	// Setting either credential config variable indicates an attempt to use configured credentials
-	if p.config.AccessKey != "" || p.config.SecretKey != "" {
+
+	if p.config.AccessKey != "" && p.config.SecretKey != "" {
```

Comment:
@Jalle19 referenced from the api:
```
  "shapes":{
    "AWSSessionCredentials":{
      "type":"structure",
      "required":[
        "accessKeyId",
        "secretAccessKey",
        "sessionToken"
      ],
      "members":{
        "accessKeyId":{"shape":"AccessKeyId"},
        "secretAccessKey":{"shape":"SecretAccessKey"},
        "sessionToken":{"shape":"SessionToken"}
      },
      "sensitive":true
    },
```
All credential values are required when initiating a new session. The sessionToken key can be an empty string if initiating a new session using STS auth which the aws-sdk-go takes care of generating if one wasn't passed into during provider creation upon initial get/retrieve.